ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/receive/view/ReceiveView.kt

```diff
@@ -170,73 +162,139 @@ private fun ReceiveContents(
 
-        Address(
-            walletAddress = walletAddress.unified,
-            onAddressCopyToClipboard = onAddressCopyToClipboard,
-            onQrImageShare = onQrImageShare,
+    Column(
+        modifier = modifier,
+    ) {
+        AddressTabs(
+            modifier = Modifier.fillMaxWidth(),
+            pagerState = pagerState,
+            tabs = state,
         )
-
-        if (versionInfo.isTestnet) {
-            Spacer(modifier = Modifier.height(ZcashTheme.dimens.spacingHuge))
-
-            Address(
-                walletAddress = walletAddress.sapling,
+        HorizontalPager(
+            modifier = Modifier.fillMaxSize(),
+            state = pagerState,
+            userScrollEnabled = false
+        ) { index ->
+            AddressPage(
+                modifier = Modifier.fillMaxSize(),
+                walletAddress = state[index],
                 onAddressCopyToClipboard = onAddressCopyToClipboard,
                 onQrImageShare = onQrImageShare,
             )
         }
+    }
+}
 
-        Spacer(modifier = Modifier.height(ZcashTheme.dimens.spacingHuge))
-
-        Address(
-            walletAddress = walletAddress.transparent,
-            onAddressCopyToClipboard = onAddressCopyToClipboard,
-            onQrImageShare = onQrImageShare,
-        )
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun AddressTabs(
+    modifier: Modifier,
+    pagerState: PagerState,
+    tabs: List<WalletAddress>,
+    coroutineScope: CoroutineScope = rememberCoroutineScope(),
+    onTabSelected: (index: Int) -> Unit = {},
+) {
+    TabRow(
+        modifier = modifier
+            .padding(horizontal = 64.dp)
+            .border(4.dp, Color.Black),
+        selectedTabIndex = pagerState.currentPage,
+        divider = {},
+        indicator = {},
+    ) {
+        tabs.forEachIndexed { index, tab ->
+            AddressTab(
+                walletAddress = tab,
+                selected = pagerState.currentPage == index,
+                onClick = {
+                    coroutineScope.launch {
+                        onTabSelected(index)
+                        pagerState.animateScrollToPage(index)
+                    }
+                },
+            )
+        }
     }
 }
 
-private val DEFAULT_QR_CODE_SIZE = 320.dp
+@Composable
+private fun AddressTab(
+    walletAddress: WalletAddress,
+    selected: Boolean,
+    onClick: () -> Unit,
+) {
+    Tab(
+        modifier = Modifier
+            .fillMaxWidth(),
+        selected = selected,
+        onClick = onClick,
+        selectedContentColor = Color.Black,
+        unselectedContentColor = Color.White
+    ) {
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .background(
+                    if (selected) Color.Black else Color.Transparent
+                )
+                .padding(vertical = 12.dp, horizontal = 2.dp),
+            contentAlignment = Alignment.Center,
+        ) {
+            Text(
```

Comment:
We could improve this a bit by making the text max two lines and ellipsis it if a longer text comes to be covered on very small devices and long locales.